.github/workflows/publish-go-tester-task.yml

```diff
@@ -132,9 +132,39 @@ jobs:
           path: ${{ env.DIST_DIR }}/${{ matrix.os.path }}
           name: ${{ matrix.os.name }}
 
+  proto-files:
+    needs:
+      - package-name-prefix
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install Task
+        uses: arduino/setup-task@v1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          version: 3.x
+
+      - name: Collect proto files
+        run: |
+          PACKAGE_NAME_PREFIX=${{ needs.package-name-prefix.outputs.prefix }}
+          export PACKAGE_NAME_PREFIX
+          task protoc:collect
+
+      - name: Upload proto files zip
+        uses: actions/upload-artifact@v3
+        with:
+          path: ${{ env.DIST_DIR }}/*.zip
+          name: rpc-protocol-files
+
```

Comment:
what do you think about adding this as an additional step in the matrix of `build` job? It seems like almost all the steps are the same, we could avoid some code duplication.

Reply:
I don't think it's semantically correct, this job doesn't build anything in itself. 
It also would require some changes to that job to make it more generic if we want to go toward that direction too.

Reply:
I wouldn't duplicate the code that way. There are some things that could be made in order to achieve that:
- If we move the task `protoc:collect` to `DistTask.yml` the workflow do not require any big change.
- another idea would be to simply rename the task as `dist:protoc-collect`
- or maybe just move `dist:` in the `matrix.os.task`

Reply:
Wasn't `DistTask.yml`shared in other projects too? I thought of putting it there but didn't cause of that.

Second and third might work but we'd still need to reword the `build` job and `os` matrix variable to be generic.